main.go

```diff
@@ -0,0 +1,231 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+
+	"encoding/json"
+
+	"github.com/Shopify/sarama"
+	"github.com/companieshouse/chs.go/avro"
+	"github.com/companieshouse/chs.go/avro/schema"
+	"github.com/companieshouse/chs.go/kafka/producer"
+	"github.com/companieshouse/kafka-management-tool/schemas"
+)
+
+// Assigns all flags to variables
+var (
+	brokerPtr         = flag.String("broker", "", "Broker address")
+	topicPtr          = flag.String("topic", "", "Topic name")
+	schemaPtr         = flag.String("schema", "", "Schema name")
+	schemaRegistryPtr = flag.String("schema-registry", "", "Schema Registry")
+	partitionPtr      = flag.Int64("partition", 0, "Partition (default: 0)")
+	offsetPtr         = flag.String("offset", "", "Offset number")
+	jsonOutPtr        = flag.Int64("json-out", 0, "Print deserialized JSON message (default: 0)")
+)
+
+func main() {
+	flag.Parse()
+
+	// check if all mandatory flags have been provided
+	validateFlags()
+
+	// Print flags and parameters of the tool
+	printFlags()
+
+	// create offset array
+	offsetArray := createOffsetArray(*offsetPtr)
+
+	// create default config for sarama
+	config := sarama.NewConfig()
+
+	//create consumer
+	consumer, err := sarama.NewConsumer([]string{*brokerPtr}, config)
+	if err != nil {
+		panic(err)
+	}
+
+	// create messages chan
+	messages := make(chan *sarama.ConsumerMessage)
+	go consumePartition(consumer, *topicPtr, int32(*partitionPtr), messages, offsetArray)
+
+	// create signals chan
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt)
+
+	consumed := 0
+ConsumerLoop:
+	for {
+		select {
+		case msg := <-messages:
+
+			// make messageBytes channel
+			messageBytes := make(chan []byte)
+
+			// if jsonOutPtr is set to 1, then output JSON and republish
+			// otherwise just republish message
+			if *jsonOutPtr == 1 {
+				go outputJSON(msg, messageBytes)
+			} else {
+				go func() {
+					messageBytes <- msg.Value
+				}()
+			}
+
+			// create producer message
+			producerMessage := &producer.Message{
+				Value: <-messageBytes,
+				Topic: *topicPtr,
+			}
+
+			//create new producer
+			p, err := producer.New(&producer.Config{Acks: &producer.WaitForAll, BrokerAddrs: []string{*brokerPtr}})
+			if err != nil {
+				fmt.Errorf("error creating producer: %v", err)
+				os.Exit(1)
+			}
+
+			// republish message
+			fmt.Printf("Republishing message for offset: %v \n", msg.Offset)
+			partition, offset, err := p.Send(producerMessage)
+			if err != nil {
+				fmt.Errorf("error republishing message with offset: %v, %v \n", msg.Offset, err)
```

Comment:
I think this won't print the error. Perhaps use panic like you have done elsewhere.

Reply:
`fmt.Printf("Republishing message for offset: %v \n", msg.Offset)` 
`panic(err)`
`os.Exit(1)`